dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowVersionDAO.java

```diff
@@ -63,4 +93,45 @@ public List<WorkflowVersion> getTagsByWorkflowIdOrderedByLastModified(long workf
         query.setMaxResults(limit);
         return query.getResultList();
     }
+
+    private List<Predicate> processQuery(String sortCol, String sortOrder, CriteriaBuilder cb, CriteriaQuery query, Root<WorkflowVersion> version) {
+        List<Predicate> predicates = new ArrayList<>();
+
+        if (!Strings.isNullOrEmpty(sortCol)) {
+            if ("open".equalsIgnoreCase(sortCol)) {
+                if ("desc".equalsIgnoreCase(sortOrder)) {
+                    query.orderBy(cb.desc(version.get("versionMetadata").get("publicAccessibleTestParameterFile")), cb.desc(version.get("id")));
+                } else {
+                    query.orderBy(cb.asc(version.get("versionMetadata").get("publicAccessibleTestParameterFile")), cb.asc(version.get("id")));
+                }
+            } else if ("descriptorTypeVersions".equals(sortCol)) {
+                if ("desc".equalsIgnoreCase(sortOrder)) {
+                    query.orderBy(cb.desc(version.get("versionMetadata").get("descriptorTypeVersions")), cb.desc(version.get("id")));
+                } else {
+                    query.orderBy(cb.asc(version.get("versionMetadata").get("descriptorTypeVersions")), cb.asc(version.get("id")));
+                }
+            } else {
+                boolean hasSortCol = version.getModel()
+                        .getAttributes()
+                        .stream()
+                        .map(Attribute::getName)
+                        .anyMatch(sortCol::equals);
```

Comment:
`equalsIgnoreCase` might be better since we don't have case-sensitive column names. And because you do a case-insensitive compare on line 101, and case-insensitive for asc/desc.